chainio/txmgr/txmgr.go

```diff
@@ -120,3 +150,56 @@ func (m *SimpleTxManager) queryReceipt(ctx context.Context, txID wallet.TxID) *t
 
 	return receipt
 }
+
+// estimateGasAndNonce we are explicitly implementing this because
+// * We want to support legacy transactions (i.e. not dynamic fee)
+// * We want to support gas management, i.e. add buffer to gas limit
+func (m *SimpleTxManager) estimateGasAndNonce(ctx context.Context, tx *types.Transaction) (*types.Transaction, error) {
+	gasTipCap, err := m.client.SuggestGasTipCap(ctx)
+	if err != nil {
+		// If the transaction failed because the backend does not support
+		// eth_maxPriorityFeePerGas, fallback to using the default constant.
+		m.log.Info("eth_maxPriorityFeePerGas is unsupported by current backend, using fallback gasTipCap")
+		gasTipCap = FallbackGasTipCap
+	}
+
+	header, err := m.client.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	gasFeeCap := new(big.Int).Add(header.BaseFee, gasTipCap)
```

Comment:
setting `gasFeeCap` to `2 * baseFee + gasTipCap` has pretty cool advantages if you want to use it: https://www.blocknative.com/blog/eip-1559-fees

Reply:
[90da821](https://github.com/Layr-Labs/eigensdk-go/pull/167/commits/90da821787ea16ea4b4f4f9343a6420213baa389)